homeassistant/components/twentemilieu/sensor.py

```diff
@@ -0,0 +1,154 @@
+            waste_type=WASTE_TYPE_ORGANIC,
+            icon="mdi:delete-empty",
+        ),
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_PAPER),
+            waste_type=WASTE_TYPE_PAPER,
+            icon="mdi:delete-empty",
+        ),
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_PLASTIC),
+            waste_type=WASTE_TYPE_PLASTIC,
+            icon="mdi:delete-empty",
+        ),
+    ]
+
+    async_add_entities(sensors, True)
+
+
+class TwenteMilieuSensor(Entity):
+    """Defines a Twente Milieu sensor."""
+
+    def __init__(
+            self,
+            twentemilieu: TwenteMilieu,
+            unique_id: str,
+            name: str,
+            waste_type: str,
+            icon: str,
+    ) -> None:
+        """Initialize the Twente Milieu entity."""
+        self._available = True
+        self._unique_id = unique_id
+        self._icon = icon
+        self._name = name
+        self._twentemilieu = twentemilieu
+        self._waste_type = waste_type
+        self._unsub_dispatcher = None
+
+        self._state = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the mdi icon of the entity."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return "_".join([DOMAIN, self._unique_id, self._waste_type])
+
+    @property
+    def should_poll(self) -> bool:
+        """Return the polling requirement of the entity."""
+        return False
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to dispatcher listening for entity data notifications."""
+        self._unsub_dispatcher = async_dispatcher_connect(
+            self.hass, DATA_UPDATE, self._schedule_immediate_update
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect from update signal."""
+        self._unsub_dispatcher()
+
+    @callback
+    def _schedule_immediate_update(self, unique_id: str) -> None:
+        """Schedule an immediate update of the entity."""
+        if unique_id == self._unique_id:
+            self.async_schedule_update_ha_state(True)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    async def async_update(self) -> None:
+        """Update Twente Milieu entity."""
+        next_pickup = await self._twentemilieu.next_pickup(self._waste_type)
+        if next_pickup is not None:
+            self._state = next_pickup.date().isoformat()
```

Comment:
Since we have a date as state, we should set the entity `device_class` to `'timestamp'`. It requires an iso formatted absolute utc timestamp. The frontend will be able to show this as a relative time.

Reply:
Will try, but I don't think it has a negative effect on the purpose of this integration.

Reply:
Set the time to a constant time, the same each date.

Reply:
Yeah, thought so. Not going there. I'm happy to adjust as soon as the system allows/shows relative time to dates and datetimes.

This will look weird ass in so many ways.

Reply:
Here are the current options:
https://www.home-assistant.io/lovelace/entities/#format

Please make the request to the frontend team.

Reply:
This PR has nothing to do with frontend, but backend. Backend wise this entities has a date. Not datetime. Where ever it is used, the data types should be correct and as is to start with. Adjusting the backend for the frontend sake, it just plain dead wrong. I'm not supporting that.

Reply:
Our internal component: Time & Date, does not do it differently btw, <https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/time_date/sensor.py>

Reply:
Sorry, I was suggesting we should ask the frontend to add this option for relative date.

I think we can keep it as absolute utc date isoformatted for now.